packages/@aws-cdk/aws-servicecatalog/lib/private/association-manager.ts

```diff
@@ -26,6 +27,31 @@ export class AssociationManager {
     };
   }
 
+  public static addEventNotifications(portfolio: IPortfolio, product: IProduct, topics: sns.ITopic[], options: TagUpdateConstraintOptions): void {
+    if (!topics.length) {
+      throw new Error(`No topics provided for launch notifications for association ${this.prettyPrintAssociation(portfolio, product)}`);
+    }
+    InputValidator.validateLength(this.prettyPrintAssociation(portfolio, product), 'description', 0, 2000, options.description);
```

Comment:
Seems like we'll do this for every association. We need to [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) this up.

Reply:
OK.